modules/internal/DefaultSparse.chpl

```diff
@@ -235,7 +235,114 @@ module DefaultSparse {
         add_help(ind);
       }
     }
-  
+
+    proc bulkAdd(inds: [] index(rank, idxType), sorted=false, noDuplicate=false){
+      var numAdded = inds.size; //maybe remove this var
+
+      //find individual insert points
+      //and eliminate duplicates between inds and dom
+      var indivInsertPts: [{0..#numAdded}] int;
+      var actualInsertPts: [{0..#numAdded}] int; //where to put in newdom
+
+      if !sorted then QuickSort(inds);
+
+      //eliminate duplicates --assumes sorted
+      if !noDuplicate {
+        //make sure lastInd != inds[inds.domain.low]
+        var lastInd = inds[inds.domain.low] + 1; 
+        for (i, p) in zip(inds, indivInsertPts)  {
+          if i == lastInd then p = -1;
```

Comment:
Actually -1 there is a marker that is not relevant to parentDom but nnzDom. And nnzDom is hardcoded to be 1-based. If I use nnzDom.low-1, the caveat is that it will be equal to 0 which is the default initial value of the array, and to avoid that there has to be an initialization to another value, which brings us to where we began. Am I missing something?